app/ClientApp/src/components/ShareRecipe.jsx

```diff
@@ -65,30 +67,58 @@ export default function ShareRecipe() {
     return true;
   };
 
+  const stringifyInstructionSteps = () => {
+    let instructions = '';
+
+    for (let i = 0; i < instructionSteps.length; i += 1) {
+      instructions = instructions.concat(`${i + 1}. ${instructionSteps[i].stepInstruction}\n`);
+    }
+    return instructions;
+  };
+
+  const calculatePrepTime = () => {
+    let fullPrepTime = 0;
+
+    for (let i = 0; i < instructionSteps.length; i += 1) {
+      fullPrepTime += instructionSteps[i].stepTime;
+    }
+
+    return fullPrepTime;
+  };
+
   const handleSubmit = async (event) => {
     event.preventDefault(); // prevents page refresh
 
     const recipeModel = {
       owner: user.email,
-      instruction: instructions,
+      instruction: stringifyInstructionSteps(),
       name: recipeTitle,
       calorieCount: 1000, // Calories and full preptime will later be calculated from all the ingredients
-      fullPrepTime: 1000,
+      fullPrepTime: calculatePrepTime(),
       imageData,
       recipeIngredient: [],
       userLikedRecipe: []
     };
+
     console.log(JSON.stringify(recipeModel));
+    console.log(JSON.stringify(ingredients));
     await authApi(user).post(`${URL}/recipe`, JSON.stringify(recipeModel));
```

Comment:
Is it better to have multiple post requests or a single one? For example, in this case, should I send everything to recipe controller and "seperate recipes and ingredients" at the backend, or should I send one request with the recipes to RecipeController and an another request to a different controller with just the ingredients?

Reply:
If they are going to the same controller, better to send one request